lib/ansible/playbook/play_context.py

```diff
@@ -49,35 +49,36 @@
 # in variable names.
 
 MAGIC_VARIABLE_MAPPING = dict(
-   connection       = ('ansible_connection',),
-   remote_addr      = ('ansible_ssh_host', 'ansible_host'),
-   remote_user      = ('ansible_ssh_user', 'ansible_user'),
-   port             = ('ansible_ssh_port', 'ansible_port'),
-   accelerate_port  = ('ansible_accelerate_port',),
-   password         = ('ansible_ssh_pass', 'ansible_password'),
-   private_key_file = ('ansible_ssh_private_key_file', 'ansible_private_key_file'),
-   pipelining       = ('ansible_ssh_pipelining', 'ansible_pipelining'),
-   shell            = ('ansible_shell_type',),
-   become           = ('ansible_become',),
-   become_method    = ('ansible_become_method',),
-   become_user      = ('ansible_become_user',),
-   become_pass      = ('ansible_become_password','ansible_become_pass'),
-   become_exe       = ('ansible_become_exe',),
-   become_flags     = ('ansible_become_flags',),
-   ssh_common_args  = ('ansible_ssh_common_args',),
-   sftp_extra_args  = ('ansible_sftp_extra_args',),
-   scp_extra_args   = ('ansible_scp_extra_args',),
-   ssh_extra_args   = ('ansible_ssh_extra_args',),
-   sudo             = ('ansible_sudo',),
-   sudo_user        = ('ansible_sudo_user',),
-   sudo_pass        = ('ansible_sudo_password', 'ansible_sudo_pass'),
-   sudo_exe         = ('ansible_sudo_exe',),
-   sudo_flags       = ('ansible_sudo_flags',),
-   su               = ('ansible_su',),
-   su_user          = ('ansible_su_user',),
-   su_pass          = ('ansible_su_password', 'ansible_su_pass'),
-   su_exe           = ('ansible_su_exe',),
-   su_flags         = ('ansible_su_flags',),
+   connection        = ('ansible_connection',),
+   remote_addr       = ('ansible_ssh_host', 'ansible_host'),
+   remote_user       = ('ansible_ssh_user', 'ansible_user'),
+   port              = ('ansible_ssh_port', 'ansible_port'),
+   accelerate_port   = ('ansible_accelerate_port',),
+   password          = ('ansible_ssh_pass', 'ansible_password'),
+   private_key_file  = ('ansible_ssh_private_key_file', 'ansible_private_key_file'),
+   pipelining        = ('ansible_ssh_pipelining', 'ansible_pipelining'),
+   shell             = ('ansible_shell_type',),
+   become            = ('ansible_become',),
+   become_method     = ('ansible_become_method',),
+   become_user       = ('ansible_become_user',),
+   become_pass       = ('ansible_become_password','ansible_become_pass'),
+   become_exe        = ('ansible_become_exe',),
+   become_flags      = ('ansible_become_flags',),
+   ssh_common_args   = ('ansible_ssh_common_args',),
+   sftp_extra_args   = ('ansible_sftp_extra_args',),
+   scp_extra_args    = ('ansible_scp_extra_args',),
+   ssh_extra_args    = ('ansible_ssh_extra_args',),
+   nspawn_extra_args = ('ansible_nspawn_extra_args',),
```

Comment:
this really does not need to be here, the connection plugin itself can look for it